### PR TITLE
D8/9 - iats recur payment test

### PR DESCRIPTION
**Pull request by jitendrapurohit**
Overview
----------------------------------------
Test recur payment with iATS. Covers the fix done in https://github.com/colemanw/webform_civicrm/pull/611

Before
----------------------------------------
No test.

After
----------------------------------------
Test Added.

### CODE REVIEW

**Comment by KarinG** [comment]
I've rebased this PR and pulling it into my local now to have a closer look.

File: tests/src/FunctionalJavascript/ContributionIatsTest.php [hunk 1]
```diff
@@ -82,22 +80,16 @@ public function testSubmit1stPayContribution() {
...
-    $this->getSession()->getPage()->selectFieldOption('Financial Type', 'Donation');
-    // throw new \Exception(var_export($this->getOptions('Payment Processor'), TRUE));
-    $this->assertCount(4, $this->getOptions('Payment Processor'));
-    $this->getSession()->getPage()->selectFieldOption('Payment Processor',  $this->payment_processor_faps['id']);
     $this->enableBillingSection();
 
     $this->saveCiviCRMSettings();
 
+    $this->submitWebformAndVerifyPayment();
+  }
```
Comment (KarinG) [inline tests/src/FunctionalJavascript/ContributionIatsTest.php:90]: I think this needs `$isRecur` = 1 (or something)

**Comment by KarinG** [comment]
Confirming it never got to the recurring bits:

image

**Comment by KarinG** [comment]
I also don't think it enables recurring contributions? No worries. I'm going to close this PR and then take the relevant bits into the `isRecur` mapping PR. That way it can also be merged together.

image

File: tests/src/FunctionalJavascript/ContributionIatsTest.php (hunk 1 above)
Comment (jitendrapurohit) [reply tests/src/FunctionalJavascript/ContributionIatsTest.php:90]: Hmm, this param was added for `testRecurringPayment()` which was removed in the second commit added to this PR? https://github.com/colemanw/webform_civicrm/pull/613/commits/e21e3edb17c9f3020984e603e89f6b56ba8a6667. The main commit has 2 tests calling this function. The above call is from `testSubmit1stPayContribution()` which is meant to do non-recur payment as it did before this PR. @KarinG

**Comment by KarinG** [comment]
Closing this - I reworked the test in the next PR

**Pull request closed without merging**